docs/features/interceptors.md

```diff
@@ -68,20 +68,13 @@ File-local declarations of this type (`file class InterceptsLocationAttribute`)
 
 #### File paths
 
-File paths used in `[InterceptsLocation]` are expected to have `/pathmap` substitution already applied. Generators should accomplish this by locally recreating the file path transformation performed by the compiler:
+The *referenced syntax tree* of an `[InterceptsLocationAttribute]` is determined using the following rules:
+1. A *mapped path* of each syntax tree is determined by applying [`/pathmap`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.commandlinearguments.pathmap?view=roslyn-dotnet-4.7.0) substitution to `SyntaxTree.FilePath`.
+2. For a given `[InterceptsLocation]` usage, the `filePath` argument value is compared to the *mapped path* of each syntax tree using ordinal string comparison. If exactly one syntax tree matches under this comparison, that is the *referenced syntax tree*.
+3. If no syntax tree matched by the above comparison, then the `filePath` argument value is resolved relative to path of the containing syntax tree of the `[InterceptsLocation]` usage. Pathmap substitution is applied to the result of this relative resolution. If exactly one syntax tree matches this resolved and mapped path, that is the *referenced syntax tree*.
+4. If neither of the above methods of comparison matched exactly one syntax tree, an error occurs.
 
-```cs
-using Microsoft.CodeAnalysis;
-
-string GetInterceptorFilePath(SyntaxTree tree, Compilation compilation)
-{
-    return compilation.Options.SourceReferenceResolver?.NormalizePath(tree.FilePath, baseFilePath: null) ?? tree.FilePath;
-}
-```
-
-The file path given in the attribute must be equal by ordinal comparison to the value given by the above function.
-
-The compiler does not map `#line` directives when determining if an `[InterceptsLocation]` attribute intercepts a particular call in syntax.
+Note that `#line` directives are not considered when determining the call referenced by an `[InterceptsLocation]` attribute.
```

Comment:
The doc still says that we consider mapped path here, thought we decided it would just match `#line`

Reply:
The term mapped path is used in reference to `/pathmap`, not `#line` mapping. This sentence is meant to say that presence or absence of `#line` doesn't affect what an `InterceptsLocationAttribute` means.

I think this is something which will become much simpler conceptually as we phase out the "compat" lookup. Basically, there will no longer be any expectation for the user of `[InterceptsLocation]` to perform any kind of mapping of the file path, line, or column; rather, to just get and use a relative path from the "current file" to the "referenced file".

Reply:
Point (2) still says that path map impacts how we resolve paths in `InterceptsLocation`. That means the following paths do not _necessarily_ mean the same thing. 

```csharp
#line /src/blah/code.cs
[Intercepts("/src/blah/code.cs")]
```

That feels very broken. Both of these features are about mapping a file path, relative or absolute, to another file in the compilation. Why should one use `/pathmap` and the other not use it? 

If I'm incorrect and `#line` does respect path map for embedding I'm happy to be corrected on this point.